Make logs in inform-re-failed-alter-strategem compatible with browser

kamalaknn opened the pull request.

kamalaknn committed:
Make logs in inform-re-failed-alter-strategem compatible with browser

kamalaknn commented:
@mikermcneil I've kept the changes minimal for quick review. Let me know if this works.

mikermcneil commented:
@kamalaknn looks great! Would you mind inlining the logic instead of creating a new function? (Prefer to avoid naming and extrapolating a function unless something’s being used in 3 separate places)

kamalaknn committed:
Inline logger for inform-re-failed-alter-strategem

kamalaknn commented:
I've updated the PR inlining the early return for browsers. I had to duplicate the done callback because of the sync vs async style inconsistency between `console.log` and `fsx.outputFile`.

mikermcneil commented:
@kamalaknn ah the curses of pre-`await` asynchronous JavaScript... (Can't wait until we don't have to support Node 6 anymore!)

Thanks!

mikermcneil committed:
add note about fs-extra up top in the imports

mikermcneil committed:
Update inform-re-failed-alter-stratagem.js

mikermcneil committed:
Deduplicate setTimeout+outlet using IIFE (see https://github.com/mikermcneil/parley/tree/a2a851e3f4f7022d7289c8f4600e5d0a236a8b35#what-if-im-stuck-with-an-old-version-of-nodejs-3 for example)

@kamalaknn btw here's how I usually deal with this stuff when we can't use `await`, in case it's helpful: 
 https://github.com/mikermcneil/parley/tree/a2a851e3f4f7022d7289c8f4600e5d0a236a8b35#what-if-im-stuck-with-an-old-version-of-nodejs-3
